server/storage/schema/cindex.go

```diff
@@ -56,32 +57,11 @@ func ReadConsistentIndex(tx backend.ReadTx) (uint64, uint64) {
 	return UnsafeReadConsistentIndex(tx)
 }
 
-func UnsafeUpdateConsistentIndex(tx backend.BatchTx, index uint64, term uint64, onlyGrow bool) {
+func UnsafeUpdateConsistentIndex(tx backend.BatchTx, index uint64, term uint64) {
 	if index == 0 {
 		// Never save 0 as it means that we didn't load the real index yet.
 		return
 	}
-
-	if onlyGrow {
-		oldi, oldTerm := UnsafeReadConsistentIndex(tx)
-		if term < oldTerm {
-			return
-		}
-		if index > oldi {
```

Comment:
Optional: I would consider being defensive. Something like this: 

```
verify.Assert(func() {  
   oldi, _ := UnsafeReadConsistentIndex(tx)
   if index <= oldi {
     lg.Fatalf("Attempt to apply transaction at <= index detected...");
   } 
 } );
```
- This shouldn't be implemented on Term (maybe in 3.6)
- This should clearly 'fail'/'panic' instead of failing silently.

Reply:
makes sense, will prepare a PR.